TCAT/NotificationBannerView.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  NotificationBannerView.swift
+//  TCAT
+//
+//  Created by HAIYING WENG on 11/13/19.
+//  Copyright © 2019 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+enum NotificationType {
+    
+    case beforeBoardingConfirmation, busArriving, busDelay, delayConfirmation
+    
+    var bannerColor: UIColor {
+        switch self {
+        case .beforeBoardingConfirmation, .busArriving, .delayConfirmation:
+            return Colors.tcatBlue
+        case .busDelay:
+            return Colors.lateRed
+        }
+    }
+
+}
+
+class NotificationBannerView: RoundShadowedView {
+
+    private var type: NotificationType
+
+    private let notificationLabel = UILabel()
+
+    init(busAttachment: NSTextAttachment, notificationType: NotificationType) {
+        let cornerRadius: CGFloat = 10
+
+        self.type = notificationType
+        super.init(cornerRadius: cornerRadius)
+
+        layer.shadowOpacity = 0.8
+
+        containerView.backgroundColor = type.bannerColor
+
+        notificationLabel.attributedText = formatTitleLabel(attachment: busAttachment)
+        notificationLabel.font = .getFont(.regular, size: 14)
+        notificationLabel.textColor = Colors.white
+        notificationLabel.lineBreakMode = .byWordWrapping
+        notificationLabel.numberOfLines = 0
+        containerView.addSubview(notificationLabel)
+        
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        let containerViewInset = 8
+        let notificationLabelInset = 15
+        let topPadding = 5
+
+        containerView.snp.remakeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(containerViewInset)
+            make.top.equalToSuperview().inset(topPadding)
+        }
+
+        notificationLabel.snp.makeConstraints { make in
+            make.edges.equalToSuperview().inset(notificationLabelInset)
+        }
+    }
+
+    private func formatTitleLabel(attachment: NSTextAttachment) -> NSMutableAttributedString {
```

Comment:
can we also change the return type to `NSAttributedString` instead of `NSMutableAttributedString` so that we can ensure the caller doesn't mutate it again after calling this function